sources/packages/api/src/services/application/application.service.ts

```diff
@@ -1221,4 +1221,48 @@ export class ApplicationService extends RecordDataModelService<Application> {
       })
       .getMany();
   }
+  /**
+   * Gets active application.
+   * @param applicationId application id.
+   * @param locationId location id.
+   * @returns active application details.
+   */
+
+  async getActiveApplication(
+    applicationId: number,
+    locationId: number,
+  ): Promise<Application> {
+    return this.repo
+      .createQueryBuilder("application")
+      .select([
+        "application.applicationStatus",
+        "application.applicationNumber",
+        "offering.offeringIntensity",
+        "offering.studyStartDate",
+        "offering.studyEndDate",
+        "location.name",
+        "offering.name",
+        "pirProgram.name",
+        "pirProgram.description",
+        "student",
+        "user.firstName",
+        "user.lastName",
+      ])
+      .leftJoin("application.offering", "offering")
```

Comment:
Applications completed must have an offering. I would recommend changing it to an inner join.

Reply:
what if an application without completed status tries to access this API? it will throw a query error instead of 
![image](https://user-images.githubusercontent.com/77353155/135680151-ba876818-41c8-45f0-a8f5-c23870ee9ffa.png)
right?

Reply:
updated